### PR TITLE
Setup Poetry and Bumpversion

### PR DESCRIPTION
**Pull request by ben9809**
## Purpose
This pull request fixes problems related to documentation issues and refactors bumpversion.cfg

## Approach
The new version of sphinx-rtd-theme has been updated to avoid any issues with the docs build.
The new .readthedocs.yml file complies with the poetry package manager.
The import was shortened to build the docs without any long module names.

## Tests for New Behavior
None.

## Checklist
None.

## Learning
None

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -1,24 +1,20 @@
...
+  os: "ubuntu-22.04"
   tools:
     python: "3.8"
+  jobs:
+    post_create_environment:
+      # Install poetry
+      - pip install poetry
+    post_install:
+      # Install dependencies
+      - VIRTUAL_ENV=$READTHEDOCS_VIRTUALENV_PATH poetry install --only docs
```
Comment (Jackal08) [inline .readthedocs.yml:16]: This is an env variable? What should it be set to?

File: .github/workflows/publish-test-dist.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        run: |
+          poetry install --without docs tests
+
+      - name: Build the package
+        run: |
+          poetry build
+
+      - name: Publish to TestPyPI
+        env:
+          POETRY_PYPI_TOKEN_TESTPYPI: ${{ secrets.TEST_PYPI_API_TOKEN }}
```
Comment (Jackal08) [inline .github/workflows/publish-test-dist.yaml:36]: I need to setup and test this env variable + build. Will take me about a week.

File: .github/workflows/publish-final-dist.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        run: |
+          poetry install --without docs tests
+
+      - name: Build the package
+        run: |
+          poetry build
+
+      - name: Publish to TestPyPI
+        env:
+          POETRY_PYPI_TOKEN_PYPI: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (Jackal08) [inline .github/workflows/publish-final-dist.yaml:34]: I need to setup and test this env variable + build. Will take me about a week.

**Review by Jackal08 — COMMENTED** [review]
Will need about a week and a bit to test this env variable and that the build works correctly.

File: .readthedocs.yml (hunk 1 above)
Comment (ben9809) [reply .readthedocs.yml:16]: $READTHEDOCS_VIRTUALENV_PATH is a predefined env variable set by Read the Docs in the build environment. It's the path to the virtual env used by 'Read the Docs'  when building the project's documentation.

Comment (ben9809) [reply .readthedocs.yml:16]: https://docs.readthedocs.io/en/stable/build-customization.html

**Review by Jackal08 — APPROVED** [review]
Excellent work, thank you

**Pull request merged**